src/com/owncloud/android/providers/UsersAndGroupsSearchProvider.java

```diff
@@ -70,11 +73,32 @@
 
     public static final String AUTHORITY = UsersAndGroupsSearchProvider.class.getCanonicalName();
     public static final String ACTION_SHARE_WITH = AUTHORITY + ".action.SHARE_WITH";
+
+    public static final String CONTENT = "content";
+
     public static final String DATA_USER = AUTHORITY + ".data.user";
     public static final String DATA_GROUP = AUTHORITY + ".data.group";
+    public static final String DATA_REMOTE = AUTHORITY + ".data.remote";
 
     private UriMatcher mUriMatcher;
 
+    private static HashMap<String, ShareType> sShareTypes;
+
+    static {
+        sShareTypes = new HashMap<>();
+        sShareTypes.put(DATA_USER, ShareType.USER);
+        sShareTypes.put(DATA_GROUP, ShareType.GROUP);
+        sShareTypes.put(DATA_REMOTE, ShareType.FEDERATED);
+    }
+
+    public static ShareType getShareType(String authority) {
+
+        if (sShareTypes.containsKey(authority)) {
```

Comment:
This method may be written in a single line; check the reference for ```HashMap#get```, it returns ```null``` if the key doesn't exist.

http://developer.android.com/intl/es/reference/java/util/HashMap.html#get(java.lang.Object)